### PR TITLE
scp package uploader

### PR DESCRIPTION
**Pull request by supunkamburugamuve**
Initial implementation of scp based topology package uploader for heron. The uploader can upload the files to a machine using the scp command. Then aurora scheduler can fetch the files using the scp command again.

### CODE REVIEW

**Comment by kramasamy** [comment]
@nlu90 @objmagic - can one of you review this?

**Comment by kramasamy** [comment]
@supunkamburugamuve - just wondering if you could add a documentation page on how to use the uploader. Very similar to other pages.

**Comment by kramasamy** [comment]
@ashvina - Can you help review the PR?

**Comment by supunkamburugamuve** [comment]
@kramasamy working on the documentation.

**Comment by kramasamy** [comment]
@supunkamburugamuve - please use the documentation format like we have for other uploaders to ensure consistency.

**Comment by objmagic** [comment]
I reviewed this PR yesterday. It looks good to me.

**Comment by supunkamburugamuve** [comment]
Thanks @objmagic. @kramasamy added the documentation.

File: heron/uploaders/src/java/com/twitter/heron/uploader/scp/ScpContext.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.uploader.scp;
+
+import com.twitter.heron.spi.common.Config;
+import com.twitter.heron.spi.common.Context;
+
+public class ScpContext extends Context {
+  public static final String HERON_UPLOADER_SCP_COMMAND = "heron.uploader.scp.command";
```
Comment (ashvina) [inline heron/uploaders/src/java/com/twitter/heron/uploader/scp/ScpContext.java]: The value of this constant is not same as the one used in config file. It is missing the substring ".options". Is that ok?

Comment (supunkamburugamuve) [inline heron/uploaders/src/java/com/twitter/heron/uploader/scp/ScpContext.java]: Thanks @ashvina. It should be options. I've changed the code to reflect that.

Comment (ashvina) [inline heron/uploaders/src/java/com/twitter/heron/uploader/scp/ScpContext.java]: Thanks @supunkamburugamuve for fixing this. Do you think a test case is needed to verify the ScpContext code? For e.g. Use a test config file and verify the methods return the expected value.

**Comment by ashvina** [comment]
LGTM 👍 

If possible add a test for config verification.

**Comment by supunkamburugamuve** [comment]
Added a test for configuration.

**Comment by kramasamy** [comment]
:+1:

**Pull request merged**